src/synthesiser/Synthesiser.cpp

```diff
@@ -477,7 +477,7 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
             PRINT_BEGIN_COMMENT(out);
 
             if (!synthesiser.lookup(clear.getRelation())->isTemp()) {
-                out << "if (performIO) ";
+                out << "if (pruneImdtRels) ";
```

Comment:
@b-scholz I believe this clears also output relations, not only the intermediate ones.

Reply:
Can you provide a simple example? We need to check whether `isTemp()` is computed correctly.

Reply:
I think isTemp() only gives you temporary relations that live within a stratum (for recursive rules). Everything else is not considered temorary. In other words, temporary != imtermediate.
If you do
```
souffle a.dl -g a.cpp
```
on
```
.decl A(n:number)
.output A

.decl C(n:number)
.output C

.decl B(n:number)
.output B
.input B

B(1).
B(2).

A(N):- B(N), N>1.
C(N):- B(N), !A(N).
```
you can see that B and A are purged:
```
...
476: if (pruneImdtRels) rel_1_B->purge();
477: if (pruneImdtRels) rel_2_A->purge();
...
```